qwandaq/src/main/java/life/genny/qwandaq/graphql/ProcessData.java

```diff
@@ -32,18 +32,28 @@ public class ProcessData extends ProcessInstanceVariables {
 	private List<String> searches;
 	private List<Answer> answers;
 
-	public List<String> getSearches() {
-		return searches;
-	}
-
-	public void setSearches(List<String> searches) {
-		this.searches = searches;
-	}
+	private boolean readonly;
 
 	public ProcessData() {
 		super();
 	}
 
+	// TODO: There is a danger here that the initial defCodes set in this object may
+	// not reflect the current defCodes state.
+	@JsonbTransient
+	public List<String> getDefCodes() {
+		// split the definition code into all the defCodes for this baseEntity
+		String defCodesStr = definitionCode;
+		if (defCodesStr.startsWith("DEF_DEF_")) { // TODO: Hack to fix prefix bug
+			defCodesStr = defCodesStr.substring("DEF_".length());
+		}
+		defCodesStr = defCodesStr.replace("[", "");
+		defCodesStr = defCodesStr.replace("]", "");
+		defCodesStr = defCodesStr.replace("\"", "");
+		defCodesStr = defCodesStr.replace("_DEF_", ",DEF_");
+		return Arrays.asList(defCodesStr.split(","));
```

Comment:
this entire method could be replaced with CommonUtils.getArrayFromString(defCodesStr, str -> {
return str.replace("\_DEF\_", "DEF_");
});